datagen/src/main/scala/com/nvidia/rapids/tests/scaletest/TableGenerator.scala

```diff
@@ -0,0 +1,269 @@
+    - 10 data columns
+   */
+  private def genDData(): DataFrame = {
+    val schema = (1 to complexity).map(i => s"d_key2_$i ${randomColumnType()}").mkString(",") +
+      "," +
+      (1 to 10).map(i => s"d_data_$i ${randomColumnType()}").mkString(",")
+    val dData = dbgen.addTable("d_data", schema, dNumRows)
+    // each key should show up about 10 time, but the overlap with c_data for key group 2 should only be about 50%
+    // c_data key group uses seed range (1, 10) so here d_data uses (6, 15) to get 50% overlap
+    val numSeedsInKeyTwo = cNumRows/10
+    val keyTwoStartSeed = numSeedsInKeyTwo / 2
+    val keyTwoEndSeed = keyTwoStartSeed + numSeedsInKeyTwo
+    dData.configureKeyGroup(
+      (1 to complexity).map(i => s"d_key2_$i"),
+      CorrelatedKeyGroup(2, keyTwoStartSeed, keyTwoEndSeed),
+      FlatDistribution()
+    )
+    dData.toDF(spark)
+  }
+  /**
+  e_data: Each scale factor corresponds to 1,000,000 rows.
+    - key group 3 with 3 columns (the unique count should be about 99% of the total number of rows and overlap with b_data key group 3 by about 90%).
+    - 10 data columns
+   */
+  private def genEData(): DataFrame = {
+    val schema = "e_key3_1 string," +
+      "e_key3_2 date," +
+      "e_key3_3 long," +
+      (1 to 10).map(i => s"e_data_$i ${randomColumnType()}").mkString(",")
+    val overlapRatio = 0.9
+    val eData = dbgen.addTable("e_data", schema, eNumRows)
+    val overlapBias = Math.round(bNumRows * (1 - overlapRatio))
+    val startSeed = 1 + overlapBias
+    val endSeed = Math.round(0.99 * eNumRows) + overlapBias
+    eData.configureKeyGroup(
+      (1 to 3).map(i => s"e_key3_$i"),
+      CorrelatedKeyGroup(3, startSeed, endSeed),
+      FlatDistribution()
+    )
+    eData.toDF(spark)
+
+  }
+  /**
+  f_facts: Each scale factor corresponds to 10,000 rows
+    - key group 4
+    - f_data_low_unique_1 data column that is a long with a unique count of 5 but overlap with a_data_low_unique_1 is only 1 of the 5.
+    - f_data_low_unique_len_1 data column that is a string with variable length ranging from 1 to 5 inclusive (even distribution of lengths)
+    - f_data_row_num_1 long column which is essentially the row number for window ordering
+    - complexity/2 data columns
+   */
+  private def genFFacts(): DataFrame = {
+    val schema = "f_key4_1 int, " +
+      "f_data_low_unique_1 long, " +
+      "f_data_low_unique_len_1 string, " +
+      "f_data_row_num_1 long, " +
+      (1 to Math.round(complexity/2.0).toInt).map(i => s"f_data_$i ${randomColumnType()}").mkString(",")
+    val fFact = dbgen.addTable("f_facts", schema, fNumRows)
+    fFact("f_key4_1").setSeedRange(1, 5)
+    // overlap only 1 of 5 => this (5, 9) vs. aFact (1, 5)
+    fFact("f_data_low_unique_1").setSeedRange(5, 9)
+    fFact("f_data_low_unique_len_1").setLength(1, 5)
+    fFact("f_data_row_num_1").setValueRange(1, fNumRows)
```

Comment:
Sorry this one also needs to be RowNumberPassThrough

Reply:
Fixed.